tests/protocol/LiquidationBlockingFix.test.js

```diff
@@ -0,0 +1,1283 @@
+
+            // Malicious borrower opens margin trade with WRBTC
+            const tx = await maliciousBorrower.performMarginTrade(
+                loanTokenWRBTC.address,
+                "0x0",
+                new BN(2).mul(oneEth),
+                loan_token_sent,
+                0,
+                SUSD.address, // collateral token
+                0,
+                "0x",
+                { value: 0 }
+            );
+
+            const decode = decodeLogs(tx.receipt.rawLogs, LoanOpenings, "Trade");
+            const loanId = decode[0].args["loanId"];
+
+            // Wait for loan to expire
+            await increaseTime(28 * 24 * 60 * 60 + 3600); // 28 days + 1 hour
+
+            // Get initial FeeSharingCollector balance
+            const initialBalance = await web3.eth.getBalance(feeSharingCollector.address);
+
+            // Get loan info before rollover
+            const loanBefore = await sovryn.getLoan(loanId);
+            console.log("Loan before rollover - Principal:", loanBefore.principal.toString());
+            console.log("Loan before rollover - Collateral:", loanBefore.collateral.toString());
+
+            // Attempt rollover - should succeed despite malicious borrower
+            const rolloverTx = await sovryn.rollover(loanId, "0x", { from: accounts[4] });
+
+            console.log("✅ Rollover succeeded despite malicious borrower");
+
+            // Get loan info after rollover
+            const loanAfter = await sovryn.getLoan(loanId);
+            console.log("Loan after rollover - Principal:", loanAfter.principal.toString());
+            console.log("Loan after rollover - Collateral:", loanAfter.collateral.toString());
+
+            // Check if donation event was emitted (may or may not happen depending on rollover scenario)
+            const donationDecode = decodeLogs(
+                rolloverTx.receipt.rawLogs,
+                LoanClosingsEvents,
+                "DonateToFeeSharingCollector"
+            );
+
+            if (donationDecode.length > 0) {
+                console.log("✅ Donation to FeeSharingCollector event emitted during rollover");
+                console.log("Original recipient:", donationDecode[0].args.originalRecipient);
+                console.log("Donated amount:", donationDecode[0].args.amount.toString());
+
+                // Verify FeeSharingCollector received the donation
+                const finalBalance = await web3.eth.getBalance(feeSharingCollector.address);
+                const balanceIncrease = new BN(finalBalance).sub(new BN(initialBalance));
+                expect(balanceIncrease).to.be.greaterThan(new BN(0));
+                console.log("FeeSharingCollector balance increase:", balanceIncrease.toString());
+            } else {
+                console.log(
+                    "ℹ️  No donation occurred during rollover (this is normal if no WRBTC was sent to malicious borrower)"
+                );
```

Comment:
if you simulate revert on refund when rollover closes a loan, the test should pass if there is a donation and fail otherwise.

Reply:
updated the [test](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/559/commits/471b16c305774b112b8b306c32d5ff27417aa10c)